src/script/sign.h

```diff
@@ -53,12 +53,19 @@ class MutableTransactionSignatureCreator : public BaseSignatureCreator {
 /** A signature creator that just produces 72-byte empty signatures. */
 extern const BaseSignatureCreator& DUMMY_SIGNATURE_CREATOR;
 
+typedef std::pair<CPubKey, std::vector<unsigned char>> SigPair;
+
 struct SignatureData {
-    CScript scriptSig;
-    CScriptWitness scriptWitness;
+    bool complete = false; // Stores whether the scriptSig and scriptWitness are complete
+    CScript scriptSig; // The scriptSig of an input. Contains complete signatures or the traditional partial signatures format
+    CScript redeem_script; // The redeemScript (if any) for the input
+    CScript witness_script; // The witnessScript (if any) for the input
+    CScriptWitness scriptWitness; // The scriptWitness of an input. Contains complete signatures or the traditional partial signatures format
+    std::map<CKeyID, SigPair> signatures; // BIP 174 style partial signatures for the input. May contain complete signatures
```

Comment:
may want to rephrase or define "complete signatures". I assume you mean it in the PSBT sense.

Reply:
Done